Source/WebCore/testing/Internals.cpp

```diff
@@ -4437,13 +4433,10 @@ bool Internals::isPluginSnapshotted(Element&)
 
 void Internals::initializeMockMediaSource()
 {
-#if USE(AVFOUNDATION)
-    WebCore::DeprecatedGlobalSettings::setAVFoundationEnabled(false);
-#endif
-#if USE(GSTREAMER)
-    WebCore::DeprecatedGlobalSettings::setGStreamerEnabled(false);
-#endif
-    MediaPlayerFactorySupport::callRegisterMediaEngine(MockMediaPlayerMediaSource::registerMediaEngine);
+    Document* document = contextDocument();
+    if (!document || !document->page())
+        return;
+    document->page()->setMockMediaSourceEnabled();
```

Comment:
If we add it there, we reuse the same GPUProcess and WebProcess for other tests, so the mock might be enabled even though it was not supposed to be enabled.
This is a pre-existing issue so maybe this is fine.
If we want to fix this issue, one possibility would be to move this API to test runner.
Then test runner would kill the GPUProcess when resetting the test options for the next test if mock MSE is enabled.

Reply:
It will get reset upon the next run as enabling the MockMSE in the GPU Process doesn't really impact what is happening in the content process.
Setting the pref that enable or disable the AVFoundation MediaPlayer will automatically reset the available MediaPlayer list in the content process.

With the RemoteMediaPlayer, the content process sends which MediaPlayer to use to the GPU Process. If it's disabled in the current content process, that's where the filtering is happening and as such attempting to open a RemoteMediaPlayer will fail.

I can run a call to disable it in the GPU Process if need be, but in practice it makes no difference.

Reply:
on 2nd thought, it we enable the MockMediaSource in the GPU process by adding the MockMediaPlayerMediaSource to the list of supported MediaPlayer, and we expect tests and content processes to be running concurrently, we don't want to do what we are doing in the CP when enabling the MockMediaSource: which is to reset the list of supported MediaPlayers and only add the MockMediaPlayerMediaSource ; We only want to to add it to the existing supported list.

Otherwise from that point on, only MockMediaSource is usable, in all content process regardless of what the test is doing.

Considering what I wrote above (that only what is allowed in the CP matters) I'll remove all code that would otherwise reset the MediaPlayer list.